Views/Purchases/Create.cshtml

```diff
@@ -44,4 +71,195 @@
+        let purchaseTotalLabel = document.getElementById('purchaseTotal');
+        let newProductId = document.getElementById('ProductId');
+        let newProductName = document.getElementById('ProductName');
+        let newQuantity = document.getElementById('ProductQuantity');
+        let newPrice = document.getElementById('ProductPrice');
+        let newTotalItem = document.getElementById('PurchaseTotalItem');
+        let newquantityInStock;
+        let productTableRowNumber = document.getElementById('ProductTableRowNumber');
+        var table = document.getElementById('PurchaseItemsTable');
+
+        var tableRowsNumber = 1;
+        let purchaseItems = [];
+        let purchaseTotal = 0;
+
+        productTableRowNumber.innerHTML = tableRowsNumber;
+
+        function addPurchaseItem() {
+            let quantityItem = parseInt(newQuantity.value);
+            let currentPriceItem = parseFloat(newPrice.value);
+
+            if (Number.isNaN(currentPriceItem) || currentPriceItem == "" || currentPriceItem === null || currentPriceItem <= 0) {
+                alert("Please, select the product.");
+                return false;
+            }
+            
+
+            const productExists = purchaseItems.some(product => product.ProductId == newProductId.value);
+            if (productExists) {
+                alert("Product already ordered.");
+                return false;
+            }
+
+            let purchaseTotalItem = quantityItem * currentPriceItem;
+            purchaseTotal += purchaseTotalItem;
+
+            var row = table.insertRow(tableRowsNumber);
+            var cell1 = row.insertCell(0);
+            var cell2 = row.insertCell(1);
+            var cell3 = row.insertCell(2);
+            var cell4 = row.insertCell(3);
+            var cell5 = row.insertCell(4);
+            var cell6 = row.insertCell(5);
+            cell1.innerHTML = tableRowsNumber;
+            cell2.innerHTML = newProductName.value;
+            cell3.innerHTML = currentPriceItem.toFixed(2);
+            cell4.innerHTML = quantityItem;
+            cell5.innerHTML = purchaseTotalItem.toFixed(2);
+            cell6.innerHTML = "<button type='button' onclick=\"removePurchaseItem(\'" + tableRowsNumber + "\')\" >Delete</button>";
+
+            purchaseTotalLabel.innerHTML = purchaseTotal.toFixed(2);
+
+            purchaseItems.push({
+                ProductId: parseInt(newProductId.value),
+                ProductQuantity: quantityItem,
+                ProductCurrentPrice: currentPriceItem,
+                Total: purchaseTotalItem,
+                DiscountTotal: 0
+            });
+
+            tableRowsNumber += 1;
+
+            productTableRowNumber.innerHTML = tableRowsNumber;
+            newProductName.value = "";
+            newPrice.value = "";
+            newQuantity.value = "";
+            newTotalItem.textContent = "";
+
+        };
+
+        function removePurchaseItem(id) {
+
+            var index = id - 1;
```

Comment:
use `let` and `const` varieble declaration to not breach scope. Use `var` only if necessary. https://stackoverflow.com/questions/762011/whats-the-difference-between-using-let-and-var

Reply:
Applied.